web/src/upload_widget.ts

```diff
@@ -34,45 +240,164 @@ export function build_widget(
     $clear_button: JQuery,
     // jQuery button to open file dialog
     $upload_button: JQuery,
+    $save_button: JQuery,
+    $scale_to_fit_button: JQuery,
+    $reset_scale_to_fit_button: JQuery,
+    cropper_options: CropperOptions,
     $preview_text?: JQuery,
     $preview_image?: JQuery,
+    $other_elements_to_hide?: JQuery,
     max_file_upload_size = default_max_file_size,
 ): UploadWidget {
-    function accept(file: File): void {
+    let first = true;
+    let reset_scale = false;
+    let edited = false;
+    let alternate_button = false;
+    let scalable = true;
+    async function accept(file: File): Promise<void> {
+        if (image_editor_options.cropperOptions && image_editor_options.actions) {
```

Comment:
- I find that a large section of code in this `accept` method is duplicated from that of in `build_direct_upload_widget`. Could you extract the similar sections into a separate method, so that we can reuse them rather than duplicating.
- Also, I notice you made most of these methods async. Is it required for these set of changes?

Reply:
Sure, we will simplify the `accept` method. 

I am not sure that we can make the code work without the async, but we will see what we can do.